src/ShopCertification/CurlRequester.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Heureka\ShopCertification;
+
+use Heureka\ShopCertification;
+
+/**
+ * @author Vladimír Kašpar <vladimir.kaspar@heureka.cz>
+ * @author Jakub Chábek <jakub.chabek@heureka.cz>
+ */
+class CurlRequester implements IRequester
+{
+
+    /**
+     * @var ApiEndpoint
+     */
+    private $endpoint;
+
+    /**
+     * @param ApiEndpoint $endpoint
+     */
+    public function setApiEndpoint(ApiEndpoint $endpoint)
+    {
+        $this->endpoint = $endpoint;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function request($action, array $data)
+    {
+        try {
+            $json = json_encode($data, JSON_PRETTY_PRINT);
```

Comment:
Tady to podle mě nikdy nemůže vyhodit exception, protože si ty data tvoříme sami ... nějaký nápad, co by to mohlo rozbít?